grpc/src/main/java/com/linecorp/armeria/server/grpc/HttpJsonTranscodingServiceBuilder.java

```diff
@@ -0,0 +1,456 @@
+            FieldMask.getDescriptor().getFullName().equals(fullName)) {
+            return JavaType.STRING;
+        }
+
+        if (isScalarValueWrapperMessage(fullName)) {
+            // "value" field. Wrappers must have one field.
+            assert messageType.getFields().size() == 1 : "Wrappers must have one 'value' field.";
+            return messageType.getFields().get(0).getJavaType();
+        }
+
+        // The messages of the following types can be sent only via HTTP body.
+        if (Struct.getDescriptor().getFullName().equals(fullName) ||
+            ListValue.getDescriptor().getFullName().equals(fullName) ||
+            Value.getDescriptor().getFullName().equals(fullName) ||
+            // google.protobuf.Any message has the following two fields:
+            //   string type_url = 1;
+            //   bytes value = 2;
+            // which look acceptable as HTTP GET parameters, but the client must send the message like below:
+            // {
+            //   "@type": "type.googleapis.com/google.protobuf.Duration",
+            //   "value": "1.212s"
+            // }
+            // There's no specifications about rewriting parameter names, so we will handle
+            // google.protobuf.Any message only when it is sent via HTTP body.
+            Any.getDescriptor().getFullName().equals(fullName)) {
+            return JavaType.MESSAGE;
+        }
+
+        return null;
+    }
+
+    private static boolean isScalarValueWrapperMessage(String fullName) {
+        return DoubleValue.getDescriptor().getFullName().equals(fullName) ||
+               FloatValue.getDescriptor().getFullName().equals(fullName) ||
+               Int64Value.getDescriptor().getFullName().equals(fullName) ||
+               UInt64Value.getDescriptor().getFullName().equals(fullName) ||
+               Int32Value.getDescriptor().getFullName().equals(fullName) ||
+               UInt32Value.getDescriptor().getFullName().equals(fullName) ||
+               BoolValue.getDescriptor().getFullName().equals(fullName) ||
+               StringValue.getDescriptor().getFullName().equals(fullName) ||
+               BytesValue.getDescriptor().getFullName().equals(fullName);
+    }
+
+    // to make it more efficient, we calculate whether extract response body one time
+    // if there is no matching toplevel field, we set it to null
+    @Nullable
+    private static String getResponseBody(List<FieldDescriptor> topLevelFields,
+                                          @Nullable String responseBody) {
+        if (StringUtil.isNullOrEmpty(responseBody)) {
+            return null;
+        }
+        for (FieldDescriptor fieldDescriptor : topLevelFields) {
+            if (fieldDescriptor.getName().equals(responseBody)) {
+                return responseBody;
+            }
+        }
+        return null;
+    }
+
+    public Map<String, GrpcMethod> methods() {
+        return methods;
+    }
```

Comment:
nit; it seems like this can be removed

```suggestion
    public Map<String, GrpcMethod> methods() {
        return methods;
    }
```